native/KotlinIOS/KotlinIOS/JourneyCell.swift

```diff
@@ -14,18 +14,33 @@ class JourneyCell: UITableViewCell {
     @IBOutlet var priceLabel: UILabel!
     @IBOutlet var departureLabel: UILabel!
     @IBOutlet var arrivalLabel: UILabel!
-    @IBOutlet var legsLabel: UILabel!
+    @IBOutlet var changesLabel: UILabel!
     
     func updateCell(_ journey: TrainTimes.Journey) {
         priceLabel.text = String(format: "£%i.%02i", journey.price / 100, journey.price % 100)
-        legsLabel.text = "\(journey.numberChanges + 1)"
+        changesLabel.text = "\(journey.numberChanges)"
+        
         
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "HH:mm:ss dd/MM/yy"
         
         let departDate: Date = Date.init(timeIntervalSince1970: TimeInterval(journey.departureTime / 1000))
         let arrivalDate = Date.init(timeIntervalSince1970: TimeInterval(journey.arrivalTime / 1000))
-        departureLabel.text = dateFormatter.string(from: departDate)
-        arrivalLabel.text = dateFormatter.string(from: arrivalDate)
+        departureLabel.text = toHumanReadableDate(departDate)
+        arrivalLabel.text = toHumanReadableDate(arrivalDate)
+    }
+    
+    func toHumanReadableDate(_ date: Date) -> String {
+        // If today: HH:mm (in xx min/hr)
+        let hourFormatter = DateFormatter()
+        hourFormatter.dateFormat = "HH:mm"
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "HH:mm dd/MM/yy"
+        
+        let now = Date()
+        if (Calendar.current.isDate(date, inSameDayAs: now)) {
```

Comment:
I think the user would find this useful even if the date is _not_ in the same day. E.g. it's currently almost midnight and the departure time is 3 hours way, which is 'the next day' but still only 3 hours away.

Reply:
That's a fair point. What would you class as close enough to show hours rather than a date?

Reply:
For times that are up to 12 hours away. Beyond that, hours are probably less useful than a time and date.

Reply:
Alright, I'll fix that